kover-gradle-plugin/docs/index.md

```diff
@@ -36,11 +42,677 @@ Gradle plugin that measures the coverage for tests running on the JVM and genera
+
+#### Total report variant
+Kover also creates a special **report variant**, which includes all classes and all tests of the module. This variant is called the **total variant** - and it does not have a special name.
+The names of the **report variants** are used when configuring the report and when running report generation tasks.
+
+#### Merging reports from different modules
+In order for classes from all necessary modules to be included in the report, as well as tests from these modules to be performed, you need to configure report aggregation.
+
+For this:
+1. select the module that will collect information about all classes and from which the report generation will be launched.
+   let's call such a module **merging module**. We recommend using the root module (root Gradle project) as a **merging module**.
+2. add dependencies in the **merging module** to all other modules whose classes should be included in the merged report
+```kotlin
+dependencies {
+    kover(project(":moduleA"))
+    kover(project(":moduleB"))
+}
+```
+
+#### Generating reports
+Reports can be generated for a single report variant, or for all classes.
+
+Suppose if we chose the root module as the merging module.
+Let's say we want to generate a report for the `custom` report:
+- To generate JaCoCo-compatible XML report, call `:koverXmlReportCustom` Gradle task
+- To generate HTML report, call `:koverHtmlReportCustom` Gradle task
+- To validate verification rule specified in project in [verification block](#specify-verification-rules-for-named-variant), call `:koverVerifyCustom` Gradle task
+- To generate binary report in IC format, call `:koverBinaryReportCustom` Gradle task
+- To print coverage to the log, call `:koverLogCustom` Gradle task
+
+*the general rule is that the name of the report variant is always added at the end
+
+Running one of these tasks automatically triggers the launch of all test tasks for `custom` report variant in all merged module.
+To exclude the automatic launch of a certain task and never take into account the coverage of this task, you should additionally [configure the project](TODO).
+
+
+If reports are needed for all classes in the merged modules, then:
+- To generate JaCoCo-compatible XML report, call `:koverXmlReport` Gradle task
+- To generate HTML report, call `:koverHtmlReport` Gradle task
+- To validate verification rule specified in project in [verification block](#specify-verification-rules-for-total-variant), call `:koverVerify` Gradle task
+- To generate binary report in IC format, call `:koverBinaryReport` Gradle task
+- To print coverage to the log, call `:koverLog` Gradle task
+
+Running one of these tasks automatically triggers the launch of all test tasks in all merged module.
+To exclude the automatic launch of a certain task and never take into account the coverage of this task, you should additionally [configure the project](TODO).
+
+#### Example
```

Comment:
I may be missing something, but this example doesn't actually have mixed JVM + android sources sets, it only has regular :app and :lib modules

Reply:
`:lib - is a JVM project, so you should write
```
kover {
    currentProject {
        createVariant("custom") {
            add("jvm")
        }
    }
}
```

to use classes from JVM target in your custom reports